.changelog/4571.bugfix.md

```diff
@@ -0,0 +1,4 @@
+RequestShutdown: fix shutdown if registration never succeeded
```

Comment:
Can this get a backport to 22.x?

Reply:
Yeah makes sense, I'll also extract this into a separate PR, as this PR will wait for andrej's PR to be merged first, so it might take some more time.